src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -3838,17 +3838,40 @@ private static bool CanReuseParameter(CSharp.Syntax.ParameterSyntax parameter)
                 type = null;
                 name = this.EatToken(SyntaxKind.ArgListKeyword);
             }
-
+            SyntaxToken exclamation;
+            SyntaxToken equals;
+            if (this.CurrentToken.Kind == SyntaxKind.ExclamationToken)
+            {
+                exclamation = this.EatToken(SyntaxKind.ExclamationToken);
+                equals = null;
+            }
+            else if (this.CurrentToken.Kind == SyntaxKind.ExclamationEqualsToken)
+            {
+                var notEq = this.EatToken(SyntaxKind.ExclamationEqualsToken);
+                equals = ConvertToMissingWithTrailingTrivia(notEq, SyntaxKind.EqualsToken);
+                equals = AddError(equals, ErrorCode.ERR_NeedSpaceBetweenExclamationAndEquals);
+                //var notEq = this.EatToken();
+                exclamation = SyntaxFactory.MissingToken(SyntaxKind.ExclamationToken);
+                //notEq = AddLeadingSkippedSyntax(notEq, SyntaxFactory.MissingToken(SyntaxKind.WhitespaceTrivia));
+                //exclamation = this.AddError(exclamation, ErrorCode.ERR_NeedSpaceBetweenExclamationAndEquals);
+            }
+            else
+            {
+                exclamation = null;
+                equals = null;
+            }
             EqualsValueClauseSyntax def = null;
             if (this.CurrentToken.Kind == SyntaxKind.EqualsToken)
             {
-                var equals = this.EatToken(SyntaxKind.EqualsToken);
+                equals = this.EatToken(SyntaxKind.EqualsToken);
+            }
+            if (!(equals is null))
```

Comment:
so the 8 or so lines above here are pretty confusing. 

1. you define EqualsvalueCluseDef, but then don't use it till later.
2. you had an else-caluse that null'ed out everything, but then you have another 'if' branch that sets 'equals'.  

I think this could definitely be cleaned up and unified more.

Reply:
thansk!